### ISSUE TITLE
User menu inaccessible to screen reader users and sighted keyboard-only users

### ISSUE THREAD

**Issue by henrycatalinismith** [henrycatalinismith/zetkin#1]
## Steps to reproduce

1. Log in
2. Attempt to focus the user menu with the tab key
3. Enable your screen reader
4. Focus the user menu

## Expected outcome

The user menu should be keyboard operable as per [WCAG success criterion 2.1.1](https://www.w3.org/WAI/WCAG21/Understanding/keyboard.html).

The role and value of the user menu should be programmatically determinable as per [WCAG success criterion 4.1.2](https://www.w3.org/WAI/WCAG21/Understanding/name-role-value.html).

## Actual outcome

The user menu isn't keyboard operable as it's not possible to move keyboard focus onto it using the the tab key. In the following GIF, the tab keypress when the logo is focused should focus the user menu. Instead focus skips over it to the signup link.

![2022-09-20 11 09 38](https://user-images.githubusercontent.com/566159/191219678-5f63d47d-e1be-4067-8b00-3085ae867b6a.gif)

In a screen reader, the user menu's role and value are ambiguous. In the following GIF, when the screen reader's focus arrives on the user menu, you would normally expect it to announce it as a button with a collapsed menu popup, but instead it's announced as plain text.

![2022-09-20 11 10 34](https://user-images.githubusercontent.com/566159/191220123-99e9bbbe-10a9-429d-b3eb-58f4917b1683.gif)

## Recommendation

* Use a `<button>` element instead of a `<div>` for the clickable element so that it'll be able to receive keyboard focus by default and will be announced as a button by screen readers.
* Add `aria-haspopup="menu"` to the button so that screen readers announce that it controls a popup menu.
* Add an `aria-expanded` attribute to the button with a value of `true` or `false` depending on the current state of the menu so that the state of the menu isn't only available visually.
* Add an `aria-label` to the button with something along the lines of `${fullName} user menu` to clarify what the menu is for, since some of its purpose is otherwise only represented visually via its top right position and the arrow icon adjacent to the name.
